src/test/java/bio/terra/service/filedata/azure/tables/TableDaoConnectedTest.java

```diff
@@ -0,0 +1,179 @@
+package bio.terra.service.filedata.azure.tables;
+
+import static bio.terra.service.common.azure.StorageTableName.DATASET_TABLE;
+import static bio.terra.service.common.azure.StorageTableName.SNAPSHOT_TABLE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.app.configuration.ConnectedTestConfiguration;
+import bio.terra.common.AzureUtils;
+import bio.terra.common.category.Connected;
+import bio.terra.common.fixtures.Names;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.filedata.FileMetadataUtils;
+import bio.terra.service.filedata.google.firestore.FireStoreDirectoryEntry;
+import bio.terra.service.snapshot.Snapshot;
+import com.azure.core.credential.AzureNamedKeyCredential;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.TableServiceClientBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "connectedtest"})
+@Category(Connected.class)
+public class TableDaoConnectedTest {
+  private final Logger logger = LoggerFactory.getLogger(TableDaoConnectedTest.class);
+  @Autowired private ConnectedTestConfiguration connectedTestConfiguration;
+  @Autowired AzureUtils azureUtils;
+  @Autowired TableDao tableDao;
+  @Autowired TableFileDao tableFileDao;
+  @Autowired TableDirectoryDao tableDirectoryDao;
+  private TableServiceClient tableServiceClient;
+  private UUID datasetId;
+  private Dataset dataset;
+  private UUID snapshotId;
+  private Snapshot snapshot;
+  private List<String> refIds;
+  private String loadTag;
+  private int numFilesToLoad;
+  private String targetBasePathFormat = "/test/%s/file-%s.json";
```

Comment:
We may see issues with parallel test runs overwriting these parent directory entries since there's only one "dataset" table 😞 Could we change "test" to a uuid?

Reply:
Great point!